src/stdio_mgr/types.py

```diff
@@ -0,0 +1,179 @@
+r"""``stdio_mgr.types`` *code module*.
+
+``stdio_mgr.types`` provides context managers for convenient
+interaction with ``stdin``/``stdout``/``stderr`` as a tuple.
+
+**Author**
+    John Vandenberg (jayvdb@gmail.com)
+
+**File Created**
+    6 Sep 2019
+
+**Copyright**
+    \(c) Brian Skinn 2018-2019
+
+**Source Repository**
+    http://www.github.com/bskinn/stdio-mgr
+
+**Documentation**
+    See README.rst at the GitHub repository
```

Comment:
> I was actually thinking about going to multiple source files.

I've been thinking about it for a while also, and `types.py` isnt enough.

Possibly

- `iotypes.py` or `io.py` for classes that *do* IO or are `IOBase` subclasses, and
- maybe `winio.py` for any Windows specific logic we may accumulate, and
- `sys.py` and/or `consoleio.py` for any logic around detecting python's console state (e.g. buffered/unbuffered detection),
- `stdiotypes.py` or `stdio.py` for the tuple context-manager class hierarchy of re-usable components, leaving
- `stdio_mgr.py` for finalised classes that implement high-level end-user-consumable tools.

Reply:
Yeah, that sort of structure makes sense to me.

> `sys.py` and/or `consoleio.py` for any logic around detecting python's console state (e.g. buffered/unbuffered detection),

Also in here any logic for dealing with cases when the `sys.stdfoo` have already been wrapped/mocked by something else?

Reply:
I have stayed with `types.py` for the tuple context-manager class hierarchy, as this is the core type hierarchy.  Many of them are not strictly for tuples, or for context managers, or for stdio ;-).

Thoughts about using `triples` in module / class names?

Reply:
> Thoughts about using `triples` in module / class names?

As in `Triple` would be the specific sub-case of `Tuple` where it contains three items? Definitely good with that.

-----

Separately, as I pore over these PRs, the value of #60 is becoming steadily clearer. I'm on board to fully type the thing.

Reply:
I started this change, but running out of time so havent finished addressing PR comments.